src/regex/match_data.cr

```diff
@@ -290,15 +290,11 @@ class Regex
     def to_s(io : IO)
       name_table = @regex.name_table
 
-      io << "#<Regex::MatchData "
-      self[0].inspect(io)
-      if size > 1
+      io << "#<Regex::MatchData"
+      size.times do |i|
         io << " "
-        (1...size).join " ", io do |i|
-          io << name_table.fetch(i) { i }
-          io << ":"
-          self[i]?.inspect(io)
-        end
+        io << name_table.fetch(i) { i } << ":" if i > 0
```

Comment:
`name_table.fetch(i) { i }` appears three times in `match_data.cr`. I would like to fix all them, so I'll create a new pull request after merging this.